platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/CustomGeometrySource.java

```diff
@@ -143,19 +145,49 @@ private void cancelTile(int z, int x, int y) {
     }
   }
 
+  @AnyThread
+  private ExecutorService obtainExecutorService() {
+    if (executorService == null || executorService.isShutdown()) {
+      createExecutorService();
+    }
+    return executorService;
+  }
+
+  @AnyThread
+  private void createExecutorService() {
```

Comment:
`fetchTile` is invoked in the native code from a thread pool and may occur on multilple threads simultaneously. Similar to the issue pointed out by @mkrussel77 in `obtainExecutorService`, this method will need to be synchronized across threads.